deployment/ccip/changeset/cs_rmn_curse_uncurse.go

```diff
@@ -0,0 +1,316 @@
+	// Bidirectional curse between two chains
+	return func(e deployment.Environment) []RMNCurseAction {
+		return append(
+			CurseLaneOnlyOnSource(sourceSelector, destinationSelector)(e),
+			CurseLaneOnlyOnSource(destinationSelector, sourceSelector)(e)...,
+		)
+	}
+}
+
+// CurseChain do a global curse on chainSelector and curse chainSelector on all other chains
+// Given 3 chains A, B, C
+// CurseChain(A) will curse A with the global curse subject and curse B and C with the curse subject of A
+func CurseChain(chainSelector uint64) CurseAction {
+	return func(e deployment.Environment) []RMNCurseAction {
+		chainSelectors := e.AllChainSelectors()
+
+		// Curse all other chains to prevent onramp from sending message to the cursed chain
+		var curseActions []RMNCurseAction
+		for _, otherChainSelector := range chainSelectors {
+			if otherChainSelector != chainSelector {
+				curseActions = append(curseActions, RMNCurseAction{
+					ChainSelector:  otherChainSelector,
+					SubjectToCurse: SelectorToSubject(chainSelector),
+				})
+			}
+		}
+
+		// Curse the chain with a global curse to prevent any onramp or offramp message from send message in and out of the chain
+		curseActions = append(curseActions, CurseGloballyOnlyOnChain(chainSelector)(e)...)
+
+		return curseActions
+	}
+}
+
+func groupRMNSubjectBySelector(rmnSubjects []RMNCurseAction, avoidCursingSelf bool, onlyKeepGlobal bool) map[uint64][]Subject {
+	grouped := make(map[uint64][]Subject)
+	for _, subject := range rmnSubjects {
+		grouped[subject.ChainSelector] = append(grouped[subject.ChainSelector], subject.SubjectToCurse)
+	}
+
+	// Only keep unique subjects, preserve only global curse if present and eliminate any curse where the chain would curse itself
+	for chainSelector, subjects := range grouped {
+		uniqueSubjects := make(map[Subject]struct{})
+		for _, subject := range subjects {
+			if subject == SelectorToSubject(chainSelector) && avoidCursingSelf {
+				continue
+			}
+			uniqueSubjects[subject] = struct{}{}
+		}
+
+		if _, ok := uniqueSubjects[GlobalCurseSubject()]; ok && onlyKeepGlobal {
+			grouped[chainSelector] = []Subject{GlobalCurseSubject()}
+		} else {
+			var uniqueSubjectsSlice []Subject
+			for subject := range uniqueSubjects {
+				uniqueSubjectsSlice = append(uniqueSubjectsSlice, subject)
+			}
+			grouped[chainSelector] = uniqueSubjectsSlice
+		}
+	}
+
+	return grouped
+}
+
+// NewRMNCurseChangeset creates a new changeset for cursing chains or lanes on RMNRemote contracts.
+// Example usage:
+//
+//	cfg := RMNCurseConfig{
+//	    CurseActions: []CurseAction{
+//	        CurseChain(SEPOLIA_CHAIN_SELECTOR),
+//	        CurseLane(SEPOLIA_CHAIN_SELECTOR, AVAX_FUJI_CHAIN_SELECTOR),
+//	    },
+//	    CurseReason: "test curse",
+//	    MCMS: &MCMSConfig{MinDelay: 0},
+//	}
+//	output, err := NewRMNCurseChangeset(env, cfg)
+func NewRMNCurseChangeset(e deployment.Environment, cfg RMNCurseConfig) (deployment.ChangesetOutput, error) {
```

Comment:
Following other Changeset naming convention. `New` is not necessary.
```suggestion
func RMNCurseChangeset(e deployment.Environment, cfg RMNCurseConfig) (deployment.ChangesetOutput, error) {
```